File: example/1d-linear-convection/weno3/julia/03b/src/flux/engquist_osher.jl
# src/flux/engquist_osher.jl
mutable struct EngquistOsherFluxCalculator <: InviscidFluxCalculator
    cfd::Any
    config::Any
    mesh::Mesh
    wave_speed::Float64
    
    function EngquistOsherFluxCalculator(cfd::Any)
        config = cfd.config
        mesh = cfd.domain.mesh
        wave_speed = config.wave_speed
        new(cfd, config, mesh, wave_speed)
    end
end

function compute!(calc::EngquistOsherFluxCalculator, qL::Vector{Float64}, qR::Vector{Float64}, flux::Vector{Float64})
    for i in 1:calc.mesh.nnodes
        c = calc.wave_speed
        cp = 0.5 * (c + abs(c))
        cm = 0.5 * (c - abs(c))
        u_L = qL[i]
        u_R = qR[i]
        flux[i] = cp * u_L + cm * u_R
    end
end

File: example/1d-linear-convection/weno3/python-v1/01f/flux/engquist_osher.py
# flux/engquist_osher.py
"""
Engquist-Osher 通量计算器（线性对流专用）
"""

from core.registry import register_component
from .base import InviscidFluxCalculator

@register_component('flux', 'engquist-osher')
class EngquistOsherFluxCalculator(InviscidFluxCalculator):
    """Engquist-Osher通量（线性对流专用）"""
    def compute(self, q_face_left, q_face_right, flux):
        for i in range(self.mesh.nnodes):
            c = self.wave_speed
            cp = 0.5 * (c + abs(c))
            cm = 0.5 * (c - abs(c))
            u_L = q_face_left[i]
            u_R = q_face_right[i]
            flux[i] = cp * u_L + cm * u_R